modules/shared_vpc_access/main.tf

```diff
@@ -63,16 +72,5 @@ resource "google_project_iam_member" "gke_host_agent" {
   count   = local.gke_shared_vpc_enabled ? 1 : 0
   project = var.host_project_id
   role    = "roles/container.hostServiceAgentUser"
-  member  = format("serviceAccount:%s", local.gke_s_account)
-}
-
-/******************************************
-  compute.networkUser role granted to dataproc service account for dataproc on shared VPC subnets
-  See: https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/network#creating_a_cluster_that_uses_a_vpc_network_in_another_project
- *****************************************/
-resource "google_project_iam_member" "dataproc_shared_vpc_network_user" {
-  count   = local.dataproc_shared_vpc_enabled ? 1 : 0
-  project = var.host_project_id
-  role    = "roles/compute.networkUser"
-  member  = format("serviceAccount:%s", local.dataproc_s_account)
+  member  = format("serviceAccount:%s", local.apis["container.googleapis.com"])
```

Comment:
We should drop the explicit per-API resources?

Reply:
We used to create this in `8.x` 
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/af1db49139c30199cd80b39eb6369d704f3ef151/modules/core_project_factory/main.tf#L469
and this is needed for GKE svpc?

Reply:
Right but let's do this in a loop for each API (same approach as we used for subnets).